oauth2client/pkce.py

```diff
@@ -0,0 +1,30 @@
+# Copyright 2016 Google Inc. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""
+Utility functions for implementing RFC7636:
+Proof Key for Code Exchange (PKCE) by OAuth Public Clients
+"""
+
+from base64 import urlsafe_b64encode
+from hashlib import sha256
+from os import urandom
+
+__author__ = 'bmccollam@uchicago.edu (Brendan McCollam)'
```

Comment:
_♬ Bear Down, Chicago Bears! ♫_

`__author__` lines are never more accurate than source control history and they are usually less accurate than source control history. Please consider omitting this line.

I'd remove all the `__author__` lines in our code today but for the fact that I'm sure I'd hurt a lot of feelings.

I probably hurt feelings when I ask for `__author__` lines to be left out of _new_ code, but I suspect I'm not hurting feelings _as much_.